sdk/exporter/exporter.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"context"
+)
+
+// Syncer is a type for functions that receive a single trace span.
+//
+// The ExportSpan method is called synchronously.
+//
+// The SpanData should not be modified.
+type Syncer interface {
```

Comment:
I'd like a longer name, `exporter.Synchronous` sounds good.

I'm conflicted over the naming in general, though. You're moving trace-specific exporter into a general-purpose package named "exporter". Now if we want to export different kinds of data, these names are claimed for trace-specific purposes. Could you add more motivation in the description?

Reply:
How about `SpanSyncer` and `SpanBatcher`? Then the possible naming conflict shouldn't happen.
I don't mind `SpanSynchronous`, though that seems unnecessarily longer now.

Reply:
IMO, exporter package should be renamed to export. export.SpanSyncer sounds better than exporter.SpanSyncer.

Reply:
I'm not 100% convinced about that, since the package for exporters themselves is named `exporter`. I've done it nevertheless.